test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Streaming/ODataJsonLightStreamWritingTests.cs

```diff
@@ -67,6 +67,79 @@ public void CanWritePrimitivePropertyValue()
+        public void CanWritePrimitivePropertyInstanceAnnotationWithoutPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CanWritePrimitivePropertyInstanceAnnotationWithPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123,\"age\":37"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                    writer.WritePrimitive(new ODataPrimitiveValue(37));
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CanWriteDynamicPropertyWithInstanceAnnotationsButWithoutPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                 ",\"aDynamicProperty@Custom.StartAnnotation\":123"
+                 );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "aDynamicProperty" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+

```

Comment:
Add tests for the following code block in the `ReadNestedPropertyInfoWithoutValue` method:
```csharp
            if ((odataAnnotations.Count + propertyAnnotations.Count) == 0)
            {
                // If we don't have any annotations for the property, it could contain errors.
                if (property != null)
                {
                    throw new ODataException(ODataErrorStrings.ODataJsonLightResourceDeserializer_PropertyWithoutValueWithWrongType(propertyName, propertyType.FullName()));
                }
                else
                {
                    // it's a dynamic property
                    throw new ODataException(ODataErrorStrings.ODataJsonLightResourceDeserializer_OpenPropertyWithoutValue(propertyName));
                }
            }
```
Part of me thinks it's dead code. I don't see how we'd get here if the reader doesn't come across an OData or custom property annotation.

Reply:
@xuzhg What's your reaction to [the above](https://github.com/OData/odata.net/pull/2786#discussion_r1400055133) comment?

Reply:
I'd like to keep those codes; I do remember It throws exception for dynamic property.

Reply:
@xuzhg If that is the case, can we add a test for that scenario?

Reply:
It's baffling though since would be in the form of `"property@custom.annotation":"#ANNOTATION_VALUE#"` or `property@odata.annotation="#ANNOTATION_VALUE#"`.
I believe even in the case of dynamic property we'd have something like `"dynamicproperty@custom.annotation":"#ANNOTATION_VALUE#"` or `dynamicproperty@odata.annotation="#ANNOTATION_VALUE#"`.

How we'd get here only to find that `odataAnnotations.Count == 0 && propertyAnnotations.Count == 0` I'm still not sure.

Reply:
I am curious about the scenario: a property:
1) without value
2) without odata instance annotation
3) without custom instance annotation

Let me remove it and see what will happen